client/my-sites/domains/domain-management/dataviews/use-context.ts

```diff
@@ -0,0 +1,227 @@
+import { DomainAction } from '@automattic/domains-table/src/domains-table/domains-table-row-actions';
+import { useDomainBulkUpdateStatus } from '@automattic/domains-table/src/use-domain-bulk-update-status';
+import { SiteExcerptData } from '@automattic/sites';
+import { useBreakpoint } from '@automattic/viewport-react';
+import { addQueryArgs } from '@wordpress/url';
+import { createContext, useCallback, useContext, useMemo, useState } from 'react';
+import { useSiteExcerptsQuery } from 'calypso/data/sites/use-site-excerpts-query';
+import { DomainData } from './use-domains';
+
+type DomainActionDescription = {
+	message?: string;
+	action: () => void | Promise< void >;
+};
+
+type OnDomainAction = (
+	action: DomainAction,
+	domain: ResponseDomain
+) => DomainActionDescription | void;
+
+interface DomainsDataViewsUpdatingDomain {
+	action: DomainAction;
+	domain: string;
+	created_at: number;
+	message?: string;
+}
+
+interface BaseDomainsDataViewsProps {
+	className?: string;
+	domains: DomainData[];
+	isLoading: boolean;
+	selectedItem: DomainData | null | undefined;
+	openDomainPane?: ( domain: DomainData ) => void;
+	isAllSitesView: boolean;
+	domainStatusPurchaseActions?: DomainStatusPurchaseActions;
+	onDomainAction?: OnDomainAction;
+	userCanSetPrimaryDomains?: boolean;
+	hideCheckbox?: boolean;
+	isLoadingDomains?: boolean;
+	useMobileCards?: boolean;
+	sidebarMode?: boolean;
+	selectedDomainName?: string;
+	createBulkAction?: ( variables: BulkUpdateVariables ) => Promise< void >;
+	fetchBulkActionStatus?: () => Promise< BulkDomainUpdateStatusQueryFnData >;
+	deleteBulkActionStatus?: () => Promise< void >;
+	currentUserCanBulkUpdateContactInfo?: boolean;
+}
+
+export type DomainsDataViewsProps =
+	| ( BaseDomainsDataViewsProps & { isAllSitesView: true } )
+	| ( BaseDomainsDataViewsProps & { isAllSitesView: false; siteSlug: string | null } );
+
+type Value = {
+	sites: Record< number, SiteExcerptData >;
+	isLoadingSites?: boolean;
+	isLoadingDomains?: boolean;
+	createBulkAction?: ( variables: BulkUpdateVariables ) => Promise< void >;
+	fetchBulkActionStatus?: () => Promise< BulkDomainUpdateStatusQueryFnData >;
+	deleteBulkActionStatus?: () => Promise< void >;
+	isAllSitesView: boolean;
+	domainStatusPurchaseActions?: DomainStatusPurchaseActions;
+	domainsRequiringAttention?: number;
+	handleAutoRenew: ( domains: DomainData[], enable: boolean ) => void;
+	handleUpdateContactInfo: ( domains: DomainData[] ) => void;
+	onDomainsRequiringAttentionChange: ( domainsRequiringAttention: number ) => void;
+	completedJobs: JobStatus[];
+	domainResults: Map< string, DomainUpdateStatus[] >;
+	handleRestartDomainStatusPolling: () => void;
+	onDomainAction( ...parameters: Parameters< OnDomainAction > ): void;
+	updatingDomain: DomainsDataViewsUpdatingDomain | null;
+	userCanSetPrimaryDomains: BaseDomainsDataViewsProps[ 'userCanSetPrimaryDomains' ];
+	isCompact: boolean;
+};
```

Comment:
I'm suggesting moving the type definition to a separate file.

Reply:
Extracted all types into separated file: client/my-sites/domains/domain-management/dataviews/types.ts

But, honestly, I'm not sure if it was a good idea. Some types just weren't supposed to be exposed...